File: Team5/GeshikaSingh/Week1/27 Oct 2020/RotateArray.cpp

class Solution {
public:
   
    void reversearray(vector<int>& nums,int start,int end){
        while(start<end){
            swap(nums[start], nums[end]);  
            start++;
            end--;
        }
    }
    
    void rotate(vector<int>& nums, int k) {
        int n=nums.size();
        k=k%n;  
        reversearray(nums,0,n-1);
        reversearray(nums,0,k-1);
        reversearray(nums,k,n-1);
    } 
};

File: Team1/Swati Basu/Rotate_Array.java
class Solution {
    public void rotate(int[] nums, int k) {
        int[] result = new int[nums.length];
        int n = nums.length;
        if(k>n){
            k = k%n;
        }
        
        for(int i=0; i<nums.length; i++)
        {
            int index = i+k;
            if(index>=nums.length)
            {index-=nums.length;}
            
            result[index] = nums[i];
        }
        System.arraycopy(result,0,nums,0,n);
        
    }
}
